clash-lib/src/Clash/Rewrite/WorkFree.hs

```diff
@@ -109,7 +109,12 @@ isWorkFree cache bndrs = go True
       Letrec bs e -> andM [go False e, allM (go False . snd) bs, allM goArg args]
       Case s _ [(_, a)] -> andM [go False s, go False a, allM goArg args]
       Case e _ _ -> andM [go False e, allM goArg args]
-      _ -> pure False
+      Cast e _ _ -> andM [go False e, allM goArg args]
+      Tick _ e -> andM [go False e, allM goArg args]
```

Comment:
`collectArgs` removes the `Tick`, so this would be in category of the `error "isWorkFree: unexpected XYZ"` alternatives.